src/main/java/com/blamejared/crafttweaker/impl/managers/GenericRecipesManager.java

```diff
@@ -0,0 +1,205 @@
+     * This allows you to explicitly specify a serializer if their name does not match the RecipeType name.
+     *
+     * @param name The recipe's resource path
+     * @param data The recipe in JSON format
+     * @docParam name "recipe_name"
+     * @docParam data {
+     * type: "minecraft:smoking",
+     * ingredient: <item:minecraft:gold_ore>,
+     * result: <item:minecraft:cooked_porkchop>,
+     * experience: 0.35 as float,
+     * cookingtime: 100
+     * }
+     */
+    @ZenCodeType.Method
+    public void addJSONRecipe(String name, IData data) {
+        if(!(data instanceof MapData)) {
+            throw new IllegalArgumentException("Json recipe's IData should be a MapData!");
+        }
+        final JsonObject recipeObject = IRecipeManager.JSON_RECIPE_GSON.fromJson(data.toJsonString(), JsonObject.class);
+        if(!recipeObject.has("type")) {
+            throw new IllegalArgumentException("Serializer type missing!");
+        }
+        if(recipeObject.get("type").getAsString().equals("crafttweaker:scripts")){
+            throw new IllegalArgumentException("Cannot add a recipe to the CraftTweaker Scripts recipe type!");
+        }
+        
+        final ResourceLocation recipeName = new ResourceLocation(CraftTweaker.MODID, name);
+        final IRecipe<?> result = RecipeManager.deserializeRecipe(recipeName, recipeObject);
+        final RecipeManagerWrapper recipeManagerWrapper = new RecipeManagerWrapper(result.getType());
+        CraftTweakerAPI.apply(new ActionAddRecipe(recipeManagerWrapper, result, null));
+    }
+    
+    @ZenCodeType.Method
+    public WrapperRecipe getRecipeByName(String name) {
+        
+        WrapperRecipe recipe = getRecipeMap().get(new ResourceLocation(name));
+        if(recipe == null) {
+            throw new IllegalArgumentException("No recipe found with name: \"" + name + "\"");
+        }
+        return recipe;
+    }
+    
+    @ZenCodeType.Method
+    public List<WrapperRecipe> getRecipesByOutput(IIngredient output) {
+    
+        return getAllRecipes().stream()
+                .filter(iRecipe -> output.matches(iRecipe.getOutput()))
+                .collect(Collectors.toList());
+    }
+    
+    @ZenCodeType.Method
+    @ZenCodeType.Getter("allRecipes")
+    public List<WrapperRecipe> getAllRecipes() {
+        
+        return getAllManagers().stream()
+                .map(IRecipeManager::getAllRecipes)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+    
+    /**
+     * Returns a map of all known recipes.
+     *
+     * @return A Map of recipe name to recipe of all known recipes.
+     */
+    @ZenCodeType.Method
+    @ZenCodeType.Getter("recipeMap")
+    public Map<ResourceLocation, WrapperRecipe> getRecipeMap() {
+        
+        return getAllManagers().stream()
+                .map(IRecipeManager::getAllRecipes)
```

Comment:
Why not using the `getRecipeMap` method directly?

Also this method does not distinguish two recipes with the same name if they happen to exist within two different recipe types. This shouldn't be possible as recipes are files on disk, but I'd mention this edge case behavior on the docs, just in case.